src/buildstream/_platform/platform.py

```diff
@@ -111,6 +111,8 @@ def canonicalize_arch(arch):
             "powerpc64le": "power-isa-le",  # Used in GCC/LLVM
             "ppc64": "power-isa-be",
             "ppc64le": "power-isa-le",
+            "riscv32": "riscv32",
```

Comment:
The latter values iirc need to be RV32I and RV64I respectively based on remote execution API feedback